torchrl/collectors/collectors.py

```diff
@@ -293,6 +298,11 @@ class SyncDataCollector(_DataCollector):
             updated. This feature should be used cautiously: if the same tensordict is added to a replay buffer for instance,
             the whole content of the buffer will be identical.
             Default is False.
+        mask_env_batch_size (Sequence[bool], optional): a sequence of bools of the same length as env.batch_size,
+            with a value of True it indicates to consider the corresponding dimension of env.batch_size as part of the
+            batch of environments used to collect frames. A value of False it indicates NOT to consider that dimension
+            as part of the batch of environments used to collect frames (used for agent dimension in multi-agent settings).
+            Default is None (corresponding to all True).
```

Comment:
```suggestion
        mask_env_batch_size (Sequence[bool], optional): a sequence of bools of the same length as env.batch_size.
            A value of ``True`` indicates that the corresponding dimension of ``env.batch_size`` is to be included in the computation of the number of frames collected. A value of ``False`` indicates NOT to consider this particular dimension
            as part of the batch of environments used to collect frames (e.g. used for agent dimension in multi-agent settings).
            Defaults to ``True`` for all dims.
```

@matteobettini 
Two questions here:
- What if we provide envs with a different batch size? (It should break when putting the tensordicts together in the sync case but it _should_ work in the async case)...
- What to do with non batch-locked envs (ie envs that are purely stateless and just read and write in a tensordict). Should we consider this case too, and constrain the `mask_env_batch_size` to have a length that match the batch dimensions of the tensordicts?

Reply:
> @matteobettini Two questions here:
> 
> * What if we provide envs with a different batch size? (It should break when putting the tensordicts together in the sync case but it _should_ work in the async case)...

The sync case will try to cat them so i think it would break. The async case will just return the one that finishes first i guess?

> * What to do with non batch-locked envs (ie envs that are purely stateless and just read and write in a tensordict). Should we consider this case too, and constrain the `mask_env_batch_size` to have a length that match the batch dimensions of the tensordicts?

In this case the mask will be checked to match env.batch_size. If during execution the batch size changes the mask will not work I guess. We could also uese a mask which is just a list of dimensions to exclude from being considered as part of the batch, something like
```python
batch_excluded_dims=[0,2]
```
Which would work as long as those dimension are part of the batch_size of the non-batch locked env.